addons-dev/aldryn-django/aldryn_django/cli.py

```diff
@@ -0,0 +1,113 @@
+#-*- coding: utf-8 -*-
+from __future__ import unicode_literals, absolute_import
+import click
+import os
+import sys
+import yaml
+from getenv import env
+from django.template import loader, Context
+from django.conf import settings as django_settings
+from aldryn_addons.utils import openfile
+
+# add the current directory to pythonpath. So the project files can be read.
+BASE_DIR = os.getcwd()
+sys.path.insert(0, BASE_DIR)
+
+settings = {}
+
+
+@click.command()
+def web():
+    """
+    launch the webserver of choice (uwsgi)
+    """
+    if any(settings[key] for key in ['ENABLE_NGINX', 'ENABLE_PAGESPEED', 'ENABLE_BROWSERCACHE']):
+        # uwsgi behind nginx. possibly with pagespeed/browsercache
+        start_with_nginx(settings)
+    else:
+        # pure uwsgi
+        execute(start_uwsgi_command())
+
+
+@click.command()
+def worker():
+    """
+    coming soon: launch the background worker
+    """
+    # TODO: celery worker startup, once available
+    pass
+
+
+@click.command()
+def migrate():
+    """
+    run any migrations needed at deploy time. most notably database migrations.
+    """
+    # TODO: create/integrate aldryn-addons migrate api
+    pass
+
+
+
+@click.group()
+def main():
+    if not os.path.exists(os.path.join(BASE_DIR, 'manage.py')):
+        raise click.UsageError('make sure you are in the same directory as manage.py')
+    from . import startup
+    startup._setup(BASE_DIR)
+    # TODO: there must be a better way
+    global settings
+    settings = {key: getattr(django_settings, key) for key in dir(django_settings)}
+
+
+main.add_command(web)
+main.add_command(worker)
+main.add_command(migrate)
+
+
+def execute(args, script=None):
+    # TODO: is cleanup needed before calling exec? (open files, ...)
+    command = script or args[0]
+    os.execvp(command, args)
+
+
+def start_uwsgi_command(port=None):
```

Comment:
port is actually not the same all the time. By default it runs on ``settings.PORT``, but if it is running behind nginx, nginx runs on ``settings.PORT`` and django runs on ``settings.BACKEND_PORT``.
Port needs to be selectively overridable here: https://github.com/aldryn/prototype-aldryn-project-template/pull/1/files#diff-aa5fb7ca5f81e7999c77f439e08e7107R100